### PR TITLE
:sparkles: change url for events

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: src/hooks/useConnectCurrentEvent.ts [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
 import { useFirestoreConnect } from "react-redux-firebase";
+import { oneHourAfterTimestamp } from "utils/time";
+import { useState, useEffect } from "react";
 import { useUser } from "./useUser";
 
 const useConnectCurrentEvent = () => {
-  const { venueId, eventId } = useParams();
+  const { venueId } = useParams();
   const { user } = useUser();
+  const [currentTimestamp, setCurrentTimestamp] = useState(0);
```
Comment (maxb94) [inline src/hooks/useConnectCurrentEvent.ts:10]: useState(Date.now() / 1000)

File: src/hooks/useConnectCurrentEvent.ts [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+
+  useEffect(() => {
+    setCurrentTimestamp(Date.now() / 1000);
+  }, []);
```
Comment (maxb94) [inline src/hooks/useConnectCurrentEvent.ts:14]: remove

File: src/pages/Account/CodeOfConduct.tsx [hunk 3]
```diff
@@ -41,11 +43,7 @@ const CodeOfConduct: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: CodeOfConductFormData) => {
     if (!user) return;
     await updateUserProfile(user.uid, data);
-    // if we use history.push, the new profile information are not in the redux store
-    // when we arrive on `venue/${venueId}/event/${eventId}` and we get redirected again to account/profile
-    window.location.assign(
-      venueId && eventId ? `venue/${venueId}/event/${eventId}` : ""
-    );
+    history.push(`/${venueId ? `v/${venueId}/live` : ""}`);
```
Comment (maxb94) [inline src/pages/Account/CodeOfConduct.tsx:46]: venueId ? `v/${venueId}/live` : ""

don' t need the outer ``

File: src/pages/Account/Login.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ const Login: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: LoginFormData) => {
     try {
       await signIn(data);
-      history.push(`/${venueId ? `venue/${venueId}${location.search}` : ""}`);
+      history.push(`/${venueId ? `v/${venueId}${location.search}` : ""}`);
```
Comment (maxb94) [inline src/pages/Account/Login.tsx:41]: same here

File: src/hooks/useConnectCurrentEvent.ts [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
 
   useFirestoreConnect([
     {
       collection: "venues",
       doc: venueId,
-      subcollections: [{ collection: "events", doc: eventId }],
+      subcollections: [{ collection: "events" }],
+      where: [
+        ["start_utc_seconds", ">=", oneHourAfterTimestamp(currentTimestamp)],
```
Comment (thecadams) [inline src/hooks/useConnectCurrentEvent.ts:22]: neat. this type of where clause and limiting would really help make the party map load faster during parties; I think we load every user in firestore on every request right now :/

File: src/pages/VenuePage/VenuePage.tsx [hunk 6]
```diff
@@ -74,11 +73,15 @@ const VenuePage = () => {
     return <>This event does not exist</>;
   }
 
-  if (!event || (event.price > 0 && !eventPurchase) || !venue || !users) {
+  if (!event || !venue || !users || !userPurchaseHistoryRequestStatus) {
```
Comment (mathieud) [inline src/pages/VenuePage/VenuePage.tsx:76]: Should we check here that the event is actually ongoing (based on the duration)?

Comment (thecadams) [reply src/pages/VenuePage/VenuePage.tsx:82]: In theory we should, but I'd be nervous about any re-render kicking someone out of the bar in the middle of their deepest philosophical moment :)

Comment (thecadams) [reply src/pages/VenuePage/VenuePage.tsx:82]: Recommendation is to not kick people out of the bar for now

File: src/pages/Account/Login.tsx [hunk 7]
```diff
@@ -38,7 +38,7 @@ const Login: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: LoginFormData) => {
     try {
       await signIn(data);
-      history.push(`/${venueId ? `venue/${venueId}${location.search}` : ""}`);
+      history.push(venueId ? `v/${venueId}${location.search}` : "");
```
Comment (mathieud) [inline src/pages/Account/Login.tsx:41]: `/v/${venueId}${location.search}`?

File: src/pages/VenuePage/VenuePage.tsx [hunk 8]
```diff
@@ -74,11 +79,16 @@ const VenuePage = () => {
...
     return <>Loading...</>;
   }
 
-  if (event.price > 0 && eventPurchaseRequestStatus && !eventPurchase) {
+  if (
+    (event.price > 0 &&
+      userPurchaseHistoryRequestStatus &&
+      !hasUserBoughtTicket) ||
+    isEventFinished
```
Comment (mathieud) [inline src/pages/VenuePage/VenuePage.tsx:90]: I would split this last one from the first 3. Event finished is different from "You're not allowed in"

**Pull request merged**